src/api/app/models/role.rb

```diff
@@ -40,42 +40,27 @@ class NotFound < APIException
 
   scope :global, -> { where(global: true) }
 
-  after_save :discard_cache
-  after_destroy :discard_cache
-
-  def self.discard_cache
-    @cache = nil
-  end
-
-  def self.rolecache
-    @cache || create_cache
-  end
-
-  def self.create_cache
-    # {"Admin" => #<Role id:1>, "downloader" => #<Role id:2>, ... }
-    @cache = Hash[Role.all.map { |role| [role.title, role] }]
+  # Fetches all roles and stores them as a hash. Uses title attribute as hash key.
+  #
+  # {"Admin" => #<Role id:1>, "downloader" => #<Role id:2>, ... }
+  def self.hashed
+    Rails.cache.fetch('hashed_roles', expires_in: 8.hours) do
```

Comment:
Too bad that the ``role`` table does not contain an ``updated_at`` column because then we could just use ``Role.all.cache_key`` as cache key and drop the ``expires_in``
https://github.com/openSUSE/open-build-service/blob/master/src/api/db/structure.sql#L933

:cry: :cry: :cry: